pallets/sto/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "256"]
+
+use codec::{Decode, Encode};
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure,
+};
+use frame_system::ensure_signed;
+use pallet_identity as identity;
+use pallet_portfolio::{self as portfolio, PortfolioAssetBalances, Trait as PortfolioTrait};
+use pallet_settlement::{
+    self as settlement, Leg, SettlementType, Trait as SettlementTrait, VenueInfo, VenueType,
+};
+use pallet_timestamp::{self as timestamp, Trait as TimestampTrait};
+use polymesh_common_utilities::{
+    constants::currency::*,
+    traits::{asset::Trait as AssetTrait, identity::Trait as IdentityTrait, CommonTrait},
+    Context,
+};
+use polymesh_primitives::{IdentityId, PortfolioId, Ticker};
+use sp_runtime::traits::{CheckedAdd, CheckedMul, Saturating};
+use sp_std::{collections::btree_set::BTreeSet, iter, prelude::*};
+
+type Identity<T> = identity::Module<T>;
+type Settlement<T> = settlement::Module<T>;
+type Timestamp<T> = timestamp::Module<T>;
+
+/// Details about the Fundraiser
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Fundraiser<Balance, Moment> {
+    /// Portfolio containing the asset being offered
+    pub offering_portfolio: PortfolioId,
+    /// Asset being offered
+    pub offering_asset: Ticker,
+    /// Asset to receive payment in
+    pub raising_asset: Ticker,
+    /// Tiers of the fundraiser.
+    /// Each tier has a set amount of tokens available at a fixed price.
+    /// The sum of the tiers is the total amount available in this fundraiser.
+    pub tiers: Vec<FundraiserTier<Balance>>,
+    /// Id of the venue to use for this fundraise
+    pub venue_id: u64,
+    /// Start of the fundraiser
+    pub start: Moment,
+    /// End of the fundraiser
+    pub end: Option<Moment>,
+    /// Fundraiser is frozen
+    pub frozen: bool,
+}
+
+/// Single tier of a tiered pricing model
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PriceTier<Balance> {
+    amount: Balance,
+    price: Balance,
+}
+
+/// Single price tier of a `Fundraiser`.
+/// Similar to a `PriceTier` but with an extra field `remaining` for tracking the amount available in a tier.
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct FundraiserTier<Balance> {
+    inner: PriceTier<Balance>,
+    remaining: Balance,
+}
+
+impl<Balance: From<u8>> Into<FundraiserTier<Balance>> for PriceTier<Balance> {
+    fn into(self) -> FundraiserTier<Balance> {
+        FundraiserTier {
+            inner: self,
+            remaining: Balance::from(0),
+        }
+    }
+}
+
+pub trait Trait:
+    frame_system::Trait
+    + CommonTrait
+    + IdentityTrait
+    + SettlementTrait
+    + TimestampTrait
```

Comment:
You don't need the timestamp pallet & common trait here as you can get that from `IdentityTrait`.